src/com/phpuaca/completion/util/PhpElementsUtil.java

```diff
@@ -0,0 +1,127 @@
+package com.phpuaca.completion.util;
+
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Condition;
+import com.intellij.patterns.ElementPattern;
+import com.intellij.patterns.PlatformPatterns;
+import com.intellij.patterns.PsiElementPattern;
+import com.intellij.psi.*;
+import com.intellij.psi.util.PsiElementFilter;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.CommonProcessors;
+import com.intellij.util.Processor;
+import com.jetbrains.php.PhpIndex;
+import com.jetbrains.php.completion.PhpLookupElement;
+import com.jetbrains.php.lang.PhpLangUtil;
+import com.jetbrains.php.lang.PhpLanguage;
+import com.jetbrains.php.lang.lexer.PhpTokenTypes;
+import com.jetbrains.php.lang.parser.PhpElementTypes;
+import com.jetbrains.php.lang.patterns.PhpPatterns;
+import com.jetbrains.php.lang.psi.PhpFile;
+import com.jetbrains.php.lang.psi.PhpPsiUtil;
+import com.jetbrains.php.lang.psi.elements.*;
+import com.jetbrains.php.lang.psi.resolve.types.PhpType;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+
+public class PhpElementsUtil {
```

Comment:
What's the logic difference between this class and PhpUtil/PhpTypeProviderUtil. I just see three similar classes with non-organized static methods.